test/integration/http-test.js

```diff
@@ -608,6 +608,73 @@ describe('HTTP APIs', function () {
         })
     })
 
+    describe('content-type-based file extensions', () => {
+      // ensure the container exists
+      before(() =>
+        server.post('/post-tests/')
+          .send(postRequest1Body)
+          .set('content-type', 'text/turtle')
+      )
+
+      describe('a new text/turtle document posted without slug', () => {
+        let response
+        before(() =>
+          server.post('/post-tests/')
+                .set('content-type', 'text/turtle; charset=utf-8')
+                .then(res => { response = res })
+        )
+
+        it('is assigned an extensionless URL', () => {
+          expect(response.headers).to.have.property('location')
+          expect(response.headers.location).to.match(/^\/post-tests\/[^./]+$/)
+        })
+      })
+
+      describe('a new text/turtle document posted with a slug', () => {
+        let response
+        before(() =>
+          server.post('/post-tests/')
+                .set('slug', 'slug1')
+                .set('content-type', 'text/turtle; charset=utf-8')
+                .then(res => { response = res })
+        )
+
+        it('is assigned an extensionless URL', () => {
+          expect(response.headers).to.have.property('location')
+          expect(response.headers.location).to.match(/^\/post-tests\/slug1+$/)
```

Comment:
Do you want that `+` before the `$`? This test will accept `/post-tests/slug111111`

Reply:
I didn't! Good catch.